base/status/StatusStatistics.ts

```diff
@@ -0,0 +1,376 @@
+                        if(parseInt(mercury) !== 0) elems_to_show.push({element: elements.MERCURY, value: mercury});
+                        if(parseInt(mars) !== 0) elems_to_show.push({element: elements.MARS, value: mars});
+                        if(parseInt(jupiter) !== 0) elems_to_show.push({element: elements.JUPITER, value: jupiter});
+
+                        for(let index in elems_to_show){
+                            if(parseInt(elems_to_show[index].value) >= 0) elems_to_show[index].value = "+" + elems_to_show[index].value;
+                            let element_name = element_names[elems_to_show[index].element];
+
+                            let line = parseInt(index) < 2 ? "line1" : "line2";
+                            msgs[line] += (parseInt(index)%2 !== 0 ? ", " : "") + element_name + " " + effect_name + " " + elems_to_show[index].value;
+                        }
+                        if(msgs.line2 === "") msgs.line1 += ".";
+                        else msgs.line2 += ".";
+                    }
+                    else{
+                        let name = (effect.key as string).toUpperCase();
+
+                        msgs = {line1: StatusStatistics.BattleBuffMsgs[name].line1,
+                            line2: StatusStatistics.BattleBuffMsgs[name].line2};
+
+                        let value = (effect.properties.values[0] ? effect.properties.values[0] : 0);
+                        msgs.line1 = msgs.line1.replace("${VALUE}", value);
+                    }
+                }
+                this.manager.update_description(msgs.line1, msgs.line2);
+            }
+        }
+    }
+
+    public on_left(){
+        let effects_count = this.manager.battle_effects_array.length;
+        
+        if(effects_count === 0) this.current_col = this.current_col === 0 ? 1 : 0;
+        else this.current_col = (this.current_col+(effects_count+1)-1)%(effects_count+1);
+
+        this.on_change();
+    }
+
+    public on_right(){
+        let effects_count = this.manager.battle_effects_array.length;
+
+        if(effects_count === 0) this.current_col = this.current_col === 0 ? 1 : 0;
+        else this.current_col = (this.current_col+1)%(effects_count+1);
+
+        this.on_change();
+    }
+
+    public on_up(){
+        if(this.current_col !== 0){
+            this.current_col = 0;
+            this.on_change();
+        }
+        else{
+            this.current_line = (this.current_line+StatusStatistics.LINES-1)%StatusStatistics.LINES;
+            this.on_change();
+        }
+    }
+
+    public on_down(){
+        if(this.current_col !== 0){
+            this.current_col = 0;
+            this.on_change();
+        }
+        else{
+            this.current_line = (this.current_line+1)%StatusStatistics.LINES;
+            this.on_change();
+        }
+    }
+
+    public initialize(){
+        let stars = ["venus_star", "mercury_star", "mars_star", "jupiter_star"];
+        for(let index in stars){
```

Comment:
[Don't use for..in to loop over arrays.](https://stackoverflow.com/questions/242841/javascript-for-in-vs-for)